lib/elixir/pages/Guards.md

```diff
@@ -0,0 +1,226 @@
+# Guards
+
+Guards are a way to augment pattern matching in a predefined set of constructs.
+
+Let's see an example of a guard used in a function clause:
+
+
+```elixir
+def long_list?(list) when length(list) > 3,  do: true
```

Comment:
Yeah in this example it doesn't shine but the point was that you can't use `[_, _, ...]` to match lists with > 1 million elements. We can change this example altogether though, to something that is not lists (e.g., using `map_size/1` which has no pattern-matching equivalent), the thing I liked about this example is that it can be achieved both with pattern matching and guards but guards may be better. We could go with `tuple_size/1` vs matching all the tuples but I feel it's not a very good example since I've probably never seen `tuple_size/1` used in the wild. 😕 Wdyt?